newhelm/typed_data.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Dict, Type, TypeVar
+from typing_extensions import Self
+from pydantic import BaseModel
+
+
+_BaseModelType = TypeVar("_BaseModelType", bound=BaseModel)
+
+
+class TypedData(BaseModel):
```

Comment:
Naming - this actually feels a bit like Protobuf's `Any`... I wonder if `AnyModel` or `ModelData` or `GenericData` would be alternatives.

I'm happy with `TypedData` though, just throwing out a few suggestions in case you like any of them better.

Reply:
I wanted to avoid "Model" as in this project readers are more likely to associate it with "ML" than Pydantic.